lightningd/runes.c

```diff
@@ -674,11 +674,17 @@ static const char *check_condition(const tal_t *ctx,
 	if (streq(alt->fieldname, "time")) {
 		return rune_alt_single_int(ctx, alt, time_now().ts.tv_sec);
 	} else if (streq(alt->fieldname, "id")) {
-		const char *id = node_id_to_hexstr(tmpctx, cinfo->peer);
-		return rune_alt_single_str(ctx, alt, id, strlen(id));
+		if (cinfo->peer) {
+			const char *id = node_id_to_hexstr(tmpctx, cinfo->peer);
+			return rune_alt_single_str(ctx, alt, id, strlen(id));
+		}
+		return rune_alt_single_missing(ctx, alt);
```

Comment:
?  Every optional field (e.g. parameter names) is the same.  They fail any test (comparison, equality, etc), *except* the "not present" test.

This cannot break any existing usage, since that always sets id.  I can't think of what I would document here?